task3_snkrs/spiders/orsay_spider.py

```diff
@@ -0,0 +1,112 @@
+import scrapy
+import w3lib.url
+from task3_snkrs.items import Item
+
+
+class OrsaySpider(scrapy.Spider):
+    name = "orsay"
+    start_urls = [
+        'https://www.orsay.com/de-de/produkte/'
+    ]
+
+    def parse(self, response):
+        urls_s = response.css('a.refinement-category-link')
+
+        return [response.follow(url_s, callback=self.parse_categories) for url_s in urls_s]
+
+    def parse_categories(self, response):
+        urls_s = response.css('a.refinement-category-link')
+
+        return [response.follow(url_s, callback=self.parse_products_url) for url_s in urls_s]
+
+    def parse_products_url(self, response):
+        urls_s = response.css('a.thumb-link')
+        yield from [response.follow(url_s, callback=self.parse_product) for url_s in urls_s]
+
+        products_found = int(response.css('.load-more-progress::attr(data-max)').get())
+        if products_found > 72:
+            pages = products_found // 72 + 1
+            return [scrapy.Request(w3lib.url.add_or_replace_parameter(response.url, 'sz', page * 72),
+                                   callback=self.parse_products_url) for page in range(2, pages)]
+
+    def parse_product(self, response):
+        item = Item()
+        item['item_id'] = self.get_item_id(response)
+        item['name'] = self.get_name(response)
+        item['description'] = self.get_description(response)
+        item['skus'] = self.get_skus(response)
+        item['image_urls'] = self.get_image_urls(response)
+        item['meta'] = self.color_requests(item, response)
+        color_requests = item['meta']
+
+        return color_requests.pop() if color_requests else item
+
+    def parse_color(self, response):
+        item = response.meta['item']
+        item['skus'] += self.get_skus(response)
+        item['image_urls'] += self.get_image_urls(response)
+        color_requests = item['meta']
+
+        return color_requests.pop() if color_requests else item
+
+    def color_requests(self, item, response):
+        color_urls = response.css('ul[class="swatches color"] li[class="selectable"] a::attr(href)').getall()
+
+        return [scrapy.Request(url, callback=self.parse_color, meta={'item': item})
+                for url in color_urls]
+
+    def get_item_id(self, response):
+        return response.css('div.product-tile::attr(data-itemid)').get()
```

Comment:
`return response.css('div.product-tile::attr(data-itemid)').getall()[0]`
Let's write it as above, so errors will be shown in logs if we are not picking product_id.